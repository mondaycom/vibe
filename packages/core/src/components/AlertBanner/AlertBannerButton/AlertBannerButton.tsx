import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import cx from "classnames";
import Button, { ButtonProps } from "../../Button/Button";
import React, { FC } from "react";
import styles from "./AlertBannerButton.module.scss";

export interface AlertBannerButtonProps extends ButtonProps {
  /**
   * If true, the button is displayed on a dark background.
   */
  isDarkBackground?: boolean;
}

const AlertBannerButton: FC<AlertBannerButtonProps> = ({
  marginLeft = false,
  isDarkBackground = false,
  id,
  "data-testid": dataTestId,
  ...buttonProps
}) => {
  const overrideButtonProps = { ...Button.defaultProps, ...buttonProps };

  const classNames = cx({
    [styles.marginLeft]: marginLeft,
    [styles.darkBackground]: isDarkBackground
  });

  return (
    <div
      className={classNames}
      data-testid={dataTestId || getTestId(ComponentDefaultTestId.ALERT_BANNER_BUTTON, id)}
      id={id}
    >
      <Button {...overrideButtonProps} size="small" className={cx(styles.bannerButton)} color="on-primary-color" />
    </div>
  );
};

Object.assign(AlertBannerButton, {
  isAlertBannerItem: true
});

export default AlertBannerButton;
