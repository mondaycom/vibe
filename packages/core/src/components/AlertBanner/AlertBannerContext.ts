import React from "react";
import { type TypographyColor } from "../Typography/Typography.types";

type AlertBannerContextType = {
  textColor: Extract<TypographyColor, "onPrimary" | "onInverted" | "fixedDark">;
};

export const AlertBannerContext = React.createContext<AlertBannerContextType>({
  textColor: "onPrimary"
});
