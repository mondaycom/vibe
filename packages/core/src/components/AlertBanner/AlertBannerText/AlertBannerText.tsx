import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import cx from "classnames";
import React, { FC, useRef } from "react";
import Tooltip from "../../Tooltip/Tooltip";
import useIsOverflowing from "../../../hooks/useIsOverflowing/useIsOverflowing";
import VibeComponentProps from "../../../types/VibeComponentProps";

import styles from "./AlertBannerText.module.scss";

export interface AlertBannerTextProps extends VibeComponentProps {
  /**
   * The text content displayed within the alert banner.
   */
  text: string;
  /**
   * If true, a left margin is applied to the alert banner text.
   */
  marginLeft?: boolean;
}

const AlertBannerText: FC<AlertBannerTextProps> = ({ text, marginLeft = false, id, "data-testid": dataTestId }) => {
  const componentRef = useRef(null);
  const classNames = cx(styles.bannerText, {
    [styles.marginLeft]: marginLeft
  });
  const isOverflowing = useIsOverflowing({ ref: componentRef });

  return (
    <Tooltip
      position="bottom"
      content={isOverflowing && text}
      showTrigger={["mouseenter"]}
      hideTrigger={["mouseleave"]}
    >
      <div
        ref={componentRef}
        className={classNames}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.ALERT_BANNER_TEXT, id)}
      >
        <span>{text}</span>
      </div>
    </Tooltip>
  );
};

Object.assign(AlertBannerText, {
  isAlertBannerItem: true,
  isAlertBannerText: true
});

export default AlertBannerText;
