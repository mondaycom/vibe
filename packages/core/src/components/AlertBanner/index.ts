export { default as AlertBanner, type AlertBannerProps } from "./AlertBanner";
export { default as AlertBannerButton, type AlertBannerButtonProps } from "./AlertBannerButton/AlertBannerButton";
export { default as AlertBannerLink, type AlertBannerLinkProps } from "./AlertBannerLink/AlertBannerLink";
export { default as AlertBannerText, type AlertBannerTextProps } from "./AlertBannerText/AlertBannerText";

export * from "./AlertBanner.types";
