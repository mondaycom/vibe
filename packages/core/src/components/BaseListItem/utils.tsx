import { EndElement, StartElement } from "./BaseListItem.types";
import { TextType } from "../Text";
import React from "react";
import Avatar from "../Avatar/Avatar";
import styles from "./BaseListItem.module.scss";
import Icon from "../Icon/Icon";
import Text from "../Text/Text";

export function renderSideElement(
  element: StartElement | EndElement,
  disabled: boolean,
  textVariant: TextType
): React.ReactNode {
  switch (element.type) {
    case "avatar":
      return (
        <Avatar
          className={styles.avatar}
          src={element.value}
          withoutBorder
          square={element.square}
          type="img"
          customSize={20}
          disabled={disabled}
        />
      );

    case "icon":
      return <Icon className={styles.icon} icon={element.value} iconSize={20} />;

    case "indent":
      return <div className={styles.indent} />;

    case "suffix":
      return (
        <Text type={textVariant} color="secondary" style={{ maxWidth: "112px" }}>
          {element.value}
        </Text>
      );

    case "custom":
      return element.render();
  }
}
