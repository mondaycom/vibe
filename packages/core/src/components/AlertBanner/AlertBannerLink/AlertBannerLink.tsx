import cx from "classnames";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import React, { FC, useContext } from "react";
import Link, { LinkProps } from "../../Link/Link";
import styles from "./AlertBannerLink.module.scss";
import { AlertBannerContext } from "../AlertBannerContext";

export interface AlertBannerLinkProps extends LinkProps {
  /**
   * If true, a left margin is applied to the link.
   */
  marginLeft?: boolean;
}

const AlertBannerLink: FC<AlertBannerLinkProps> = ({
  marginLeft = false,
  id,
  "data-testid": dataTestId,
  ...linkProps
}) => {
  const { textColor } = useContext(AlertBannerContext);
  const classNames = cx({
    [styles.marginLeft]: marginLeft
  });

  return (
    <div
      className={classNames}
      data-testid={dataTestId || getTestId(ComponentDefaultTestId.ALERT_BANNER_LINK, id)}
      id={id}
    >
      <Link
        {...linkProps}
        textClassName={cx(styles.bannerLink, {
          [styles.bannerLinkTextColorOnPrimary]: textColor === "onPrimary",
          [styles.bannerLinkTextColorOnInverted]: textColor === "onInverted"
        })}
      />
    </div>
  );
};

Object.assign(AlertBannerLink, {
  isAlertBannerItem: true
});

export default AlertBannerLink;
