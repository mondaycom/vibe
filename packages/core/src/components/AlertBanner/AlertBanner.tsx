import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import cx from "classnames";
import React, { ForwardedRef, forwardRef, ReactElement, useMemo } from "react";
import IconButton from "../../components/IconButton/IconButton";
import { CloseSmall } from "@vibe/icons";
import { AlertBannerBackgroundColor as AlertBannerBackgroundColorEnum } from "./AlertBannerConstants";
import { AlertBannerBackgroundColor } from "./AlertBanner.types";
import { NOOP } from "../../utils/function-utils";
import VibeComponentProps from "../../types/VibeComponentProps";
import { AlertBannerLinkProps } from "./AlertBannerLink/AlertBannerLink";
import { AlertBannerButtonProps } from "./AlertBannerButton/AlertBannerButton";
import { AlertBannerTextProps } from "./AlertBannerText/AlertBannerText";
import { ComponentDefaultTestId } from "../../tests/constants";
import { getTestId } from "../../tests/test-ids-utils";
import { VibeComponent, withStaticProps } from "../../types";
import styles from "./AlertBanner.module.scss";
import Text from "../Text/Text";
import { AlertBannerContext } from "./AlertBannerContext";

type ChildrenType = ReactElement<AlertBannerButtonProps | AlertBannerLinkProps | AlertBannerTextProps>;

export interface AlertBannerProps extends VibeComponentProps {
  /**
   * The background color of the alert banner.
   */
  backgroundColor?: AlertBannerBackgroundColor;
  /**
   * If true, the close button is hidden.
   */
  isCloseHidden?: boolean;
  /**
   * The ARIA label of the alert banner for accessibility.
   */
  ariaLabel?: string;
  /**
   * The ARIA label of the close button for accessibility.
   */
  closeButtonAriaLabel?: string;
  /**
   * Callback fired when the close button is clicked.
   */
  onClose?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  /**
   * The content of the alert banner.
   */
  children?: ChildrenType | ChildrenType[];
}

const AlertBanner: VibeComponent<AlertBannerProps> & {
  backgroundColors?: typeof AlertBannerBackgroundColorEnum;
} = forwardRef(
  (
    {
      children: originalChildren,
      className,
      backgroundColor = "primary",
      onClose = NOOP,
      ariaLabel = "",
      closeButtonAriaLabel = "Close",
      isCloseHidden = false,
      id,
      "data-testid": dataTestId
    }: AlertBannerProps,
    ref: ForwardedRef<HTMLDivElement>
  ) => {
    const classNames = useMemo(() => {
      return cx(className, styles.alertBanner, getStyle(styles, backgroundColor));
    }, [className, backgroundColor]);

    const isDarkBackground = backgroundColor === "dark";
    const isFixedColor = backgroundColor === "warning";
    const textColor = useMemo(() => {
      if (isFixedColor) {
        return "fixedDark";
      }
      return isDarkBackground ? "onInverted" : "onPrimary";
    }, [isDarkBackground, isFixedColor]);
    const children = useMemo(() => {
      const allChildren = React.Children.toArray(originalChildren) as ReactElement[];
      const filteredChildren = allChildren.filter(
        (
          child: ReactElement & {
            type: Record<string, unknown>;
          }
        ) => {
          if (child.type.isAlertBannerItem || child.type.displayName === "MDXCreateElement") return true;
          console.error(
            "Alert banner child is not supported. Please use AlertBannerText, AlertBannerLink or AlertBannerButton.",
            child
          );
          return false;
        }
      );

      return filteredChildren.map((child, index) => {
        return React.cloneElement(child, {
          ...child?.props,
          marginLeft: index > 0,
          isDarkBackground
        });
      });
    }, [originalChildren, isDarkBackground]);

    return (
      <Text
        type="text2"
        color={textColor}
        ref={ref}
        className={classNames}
        role="banner"
        aria-label={ariaLabel || "banner"}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.ALERT_BANNER, id)}
      >
        <AlertBannerContext.Provider value={{ textColor }}>
          <div className={cx(styles.content)}>
            {children.map(
              (
                child: ReactElement & {
                  type: Record<string, unknown>;
                },
                index
              ) => {
                const childTypeIsAlertBannerText = child.type.isAlertBannerText;
                return (
                  <div
                    key={index}
                    className={cx(styles.contentItem, {
                      [styles.contentItemText]: childTypeIsAlertBannerText
                    })}
                  >
                    {childTypeIsAlertBannerText ? <div className={cx(styles.ellipsis)}>{child}</div> : child}
                  </div>
                );
              }
            )}
          </div>
        </AlertBannerContext.Provider>
        <div className={cx(styles.closeButtonWrapper)}>
          {isCloseHidden ? null : (
            <IconButton
              data-testid="alert-banner-close-button"
              icon={CloseSmall}
              className={cx(styles.closeBtn)}
              hideTooltip
              onClick={onClose}
              size="small"
              kind="tertiary"
              color={isDarkBackground ? "on-inverted-background" : "on-primary-color"}
              ariaLabel={closeButtonAriaLabel}
            />
          )}
        </div>
      </Text>
    );
  }
);

export default withStaticProps(AlertBanner, { backgroundColors: AlertBannerBackgroundColorEnum });
