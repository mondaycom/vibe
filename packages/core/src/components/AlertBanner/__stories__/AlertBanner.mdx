import { Canvas, Meta } from "@storybook/blocks";
import { ComponentRules, Link, UsageGuidelines } from "vibe-storybook-components";
import AlertBanner from "../AlertBanner";
import AlertBannerText from "../AlertBannerText/AlertBannerText";
import AlertBannerLink from "../AlertBannerLink/AlertBannerLink";
import AlertBannerButton from "../AlertBannerButton/AlertBannerButton";
import {
  ATTENTION_BOX,
  TOAST,
  TOOLTIP
} from "../../../storybook/components/related-components/component-description-map";
import { TipWhenToUse } from "./AlertBanner.stories.helpers";
import * as AlertBannerStories from "./AlertBanner.stories";
import Flex from "../../Flex/Flex";

<Meta of={AlertBannerStories} />

# Alert Banner

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Do’s and don’ts](#dos-and-donts)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

Alert banners show pressing and high-signal messages, such as system alerts. They are meant to be noticed and prompt users to take action.

<Canvas of={AlertBannerStories.Overview} />

### Import

```js
import { AlertBanner } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Alert banners should have call for action and an option to dismiss.",
    "Don't include more than one call to action in an alert banner.",
    "Place the banner on the top of the screen, and push all other content below it."
  ]}
/>

<TipWhenToUse />

## Variants

### Types

There are five types of alert banners: primary, positive, negative, warning and inverted.

<Canvas of={AlertBannerStories.Types} />

### Alert Banner with button

<Canvas of={AlertBannerStories.AlertBannerWithButton} />

### Alert Banner with link

<Canvas of={AlertBannerStories.AlertBannerWithLink} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      positive: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner>
              <AlertBannerText text="Join us at Elevate!" />
              <AlertBannerLink text="RSVP now" href="https://monday.com" />
            </AlertBanner>
          </div>
        ),
        description: "Use banners for system messages, background processes, and general updates."
      },
      negative: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner>
              <AlertBannerText text="We successfully archived 1 item" />
            </AlertBanner>
          </div>
        ),
        description: (
          <>
            Don’t use banners for notifying a user of an action they have taken. Instead, provide visual feedback with a{" "}
            <StorybookLink page="Components/Toast">Toast.</StorybookLink>
          </>
        )
      }
    },
    {
      positive: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner backgroundColor="negative">
              <AlertBannerText text="No connection" />
              <AlertBannerLink text="Learn more" href="https://monday.com" />
              <AlertBannerButton>Try again</AlertBannerButton>
            </AlertBanner>
          </div>
        ),
        description: "If two actions are needed, use two different call to acitons."
      },
      negative: {
        component: (
          <Flex direction="column" gap={16} style={{ width: "392px" }}>
            <AlertBanner backgroundColor="negative">
              <AlertBannerText text="No connection" />
              <AlertBannerLink text="Learn more" href="https://monday.com" />
              <AlertBannerLink text="Try again" href="https://monday.com" />
            </AlertBanner>
            <AlertBanner backgroundColor="negative">
              <AlertBannerText text="No connection" />
              <AlertBannerButton>Learn more</AlertBannerButton>
              <AlertBannerButton>Try again</AlertBannerButton>
            </AlertBanner>
          </Flex>
        ),
        description: "Don't include more than one action in an alert banner with the same type."
      }
    },
    {
      positive: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner backgroundColor="dark">
              <AlertBannerText text="Join us at Elevate!" />
              <AlertBannerLink text="RSVP now" href="https://monday.com" />
            </AlertBanner>
          </div>
        ),
        description: "Use only the 4 color types: primary, negative, positive, and inverted."
      },
      negative: {
        component: (
          <div style={{ width: "392px" }}>
            <AlertBanner>
              <AlertBannerText text="Join us at Elevate!" />
              <AlertBannerLink text="RSVP now" href="https://monday.com" />
            </AlertBanner>
          </div>
        ),
        description: "Don’t choose other colors for alert banners. Keep it consistent."
      }
    }
  ]}
/>

## Use cases and examples

### Alert banner as an announcement

Use when you'd like to notify about an event or cross-company announcment.

<Canvas of={AlertBannerStories.AlertBannerAsAnAnnouncement} />

### Alert banner as an opportunity to upgrade

Use to show a trial user the number of remaining free days to use the platform.

<Canvas of={AlertBannerStories.AlertBannerAsAnOpportunityToUpgrade} />

### Overflow text

In case that there's not enough space for the content, use an ellipses (...).

<Canvas of={AlertBannerStories.OverflowText} />

## Related components

<RelatedComponents componentsNames={[TOAST, ATTENTION_BOX, TOOLTIP]} />
