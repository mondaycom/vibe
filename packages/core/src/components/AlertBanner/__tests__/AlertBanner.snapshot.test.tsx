import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import AlertBanner from "../AlertBanner";
import AlertBannerLink from "../AlertBannerLink/AlertBannerLink";
import AlertBannerButton from "../AlertBannerButton/AlertBannerButton";
import AlertBannerText from "../AlertBannerText/AlertBannerText";
import { NOOP } from "../../../utils/function-utils";

describe("AlertBanner", () => {
  it("should render correctly without props", () => {
    const tree = renderer.create(<AlertBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with props", () => {
    const tree = renderer
      .create(
        <AlertBanner
          onClose={NOOP}
          backgroundColor="negative"
          ariaLabel="lorem-banner"
          className="my-lorem-ipsum-banner"
        >
          <AlertBannerText text="Lorem ipsum dolor sit amet" />
        </AlertBanner>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with correctly with text and link", () => {
    const tree = renderer
      .create(
        <AlertBanner onClose={NOOP}>
          <AlertBannerText text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua" />
          <AlertBannerLink text="Lorem ipsum" href="https://monday.com" />
        </AlertBanner>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with correctly with text and button", () => {
    const tree = renderer
      .create(
        <AlertBanner onClose={NOOP}>
          <AlertBannerText text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua" />
          <AlertBannerButton onClick={NOOP}>Lorem Ipsum Salura</AlertBannerButton>
        </AlertBanner>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with correctly with multiple elements", () => {
    const tree = renderer
      .create(
        <AlertBanner onClose={NOOP}>
          <AlertBannerText text="Lorem ipsum dolor sit amet, consectetur adipiscing elit" />
          <AlertBannerLink text="Lorem ipsum" href="https://monday.com" />
          <AlertBannerText text="sed do eiusmod tempor incididunt ut labore et dolore magna aliqua" />
          <AlertBannerButton onClick={NOOP}>Lorem Ipsum Bailar</AlertBannerButton>
        </AlertBanner>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with hidden close button", () => {
    const tree = renderer.create(<AlertBanner isCloseHidden />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with background color", () => {
    const tree = renderer.create(<AlertBanner backgroundColor="positive" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
