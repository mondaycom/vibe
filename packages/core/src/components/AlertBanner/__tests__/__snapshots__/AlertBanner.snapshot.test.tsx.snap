// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertBanner > should render correctly with background color 1`] = `
<div
  aria-label="banner"
  className="typography onPrimary start singleLineEllipsis text text2Normal alertBanner positive"
  data-testid="alert-banner"
  role="banner"
>
  <div
    className="content"
  />
  <div
    className="closeButtonWrapper"
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      aria-label="Close"
      className="closeBtn button sizeMedium kindTertiary colorOnPrimaryColor noSidePadding"
      data-testid="alert-banner-close-button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      style={
        {
          "alignItems": "center",
          "height": "32px",
          "justifyContent": "center",
          "padding": 0,
          "width": "32px",
        }
      }
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon noFocusStyle"
        data-testid="icon"
        data-vibe="Icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 20 20"
        width="20"
      >
        <path
          d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`AlertBanner > should render correctly with hidden close button 1`] = `
<div
  aria-label="banner"
  className="typography onPrimary start singleLineEllipsis text text2Normal alertBanner primary"
  data-testid="alert-banner"
  role="banner"
>
  <div
    className="content"
  />
  <div
    className="closeButtonWrapper"
  />
</div>
`;

exports[`AlertBanner > should render correctly with props 1`] = `
<div
  aria-label="lorem-banner"
  className="typography onPrimary start singleLineEllipsis text text2Normal my-lorem-ipsum-banner alertBanner negative"
  data-testid="alert-banner"
  role="banner"
>
  <div
    className="content"
  >
    <div
      className="contentItem contentItemText"
    >
      <div
        className="ellipsis"
      >
        <div
          className="bannerText"
          data-testid="alert-banner-text"
        >
          <span>
            Lorem ipsum dolor sit amet
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="closeButtonWrapper"
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      aria-label="Close"
      className="closeBtn button sizeMedium kindTertiary colorOnPrimaryColor noSidePadding"
      data-testid="alert-banner-close-button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      style={
        {
          "alignItems": "center",
          "height": "32px",
          "justifyContent": "center",
          "padding": 0,
          "width": "32px",
        }
      }
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon noFocusStyle"
        data-testid="icon"
        data-vibe="Icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 20 20"
        width="20"
      >
        <path
          d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`AlertBanner > should render correctly without props 1`] = `
<div
  aria-label="banner"
  className="typography onPrimary start singleLineEllipsis text text2Normal alertBanner primary"
  data-testid="alert-banner"
  role="banner"
>
  <div
    className="content"
  />
  <div
    className="closeButtonWrapper"
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      aria-label="Close"
      className="closeBtn button sizeMedium kindTertiary colorOnPrimaryColor noSidePadding"
      data-testid="alert-banner-close-button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      style={
        {
          "alignItems": "center",
          "height": "32px",
          "justifyContent": "center",
          "padding": 0,
          "width": "32px",
        }
      }
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon noFocusStyle"
        data-testid="icon"
        data-vibe="Icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 20 20"
        width="20"
      >
        <path
          d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`AlertBanner > should render with correctly with multiple elements 1`] = `
<div
  aria-label="banner"
  className="typography onPrimary start singleLineEllipsis text text2Normal alertBanner primary"
  data-testid="alert-banner"
  role="banner"
>
  <div
    className="content"
  >
    <div
      className="contentItem contentItemText"
    >
      <div
        className="ellipsis"
      >
        <div
          className="bannerText"
          data-testid="alert-banner-text"
        >
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit
          </span>
        </div>
      </div>
    </div>
    <div
      className="contentItem"
    >
      <div
        className="marginLeft"
        data-testid="alert-banner-link"
      >
        <a
          aria-describedby=""
          aria-label=""
          aria-labelledby=""
          className="link colorPrimary"
          data-testid="link"
          href="https://monday.com"
          id=""
          onClick={[Function]}
          rel="noreferrer"
          target="_blank"
        >
          <span
            className="text bannerLink bannerLinkTextColorOnPrimary"
          >
            Lorem ipsum
          </span>
        </a>
      </div>
    </div>
    <div
      className="contentItem contentItemText"
    >
      <div
        className="ellipsis"
      >
        <div
          className="bannerText marginLeft"
          data-testid="alert-banner-text"
        >
          <span>
            sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
          </span>
        </div>
      </div>
    </div>
    <div
      className="contentItem"
    >
      <div
        className="marginLeft"
        data-testid="alert-banner-button"
      >
        <button
          aria-busy={false}
          aria-disabled={false}
          className="bannerButton button sizeSmall kindPrimary colorOnPrimaryColor"
          data-testid="button"
          data-vibe="Button"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          Lorem Ipsum Bailar
        </button>
      </div>
    </div>
  </div>
  <div
    className="closeButtonWrapper"
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      aria-label="Close"
      className="closeBtn button sizeMedium kindTertiary colorOnPrimaryColor noSidePadding"
      data-testid="alert-banner-close-button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      style={
        {
          "alignItems": "center",
          "height": "32px",
          "justifyContent": "center",
          "padding": 0,
          "width": "32px",
        }
      }
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon noFocusStyle"
        data-testid="icon"
        data-vibe="Icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 20 20"
        width="20"
      >
        <path
          d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`AlertBanner > should render with correctly with text and button 1`] = `
<div
  aria-label="banner"
  className="typography onPrimary start singleLineEllipsis text text2Normal alertBanner primary"
  data-testid="alert-banner"
  role="banner"
>
  <div
    className="content"
  >
    <div
      className="contentItem contentItemText"
    >
      <div
        className="ellipsis"
      >
        <div
          className="bannerText"
          data-testid="alert-banner-text"
        >
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
          </span>
        </div>
      </div>
    </div>
    <div
      className="contentItem"
    >
      <div
        className="marginLeft"
        data-testid="alert-banner-button"
      >
        <button
          aria-busy={false}
          aria-disabled={false}
          className="bannerButton button sizeSmall kindPrimary colorOnPrimaryColor"
          data-testid="button"
          data-vibe="Button"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          Lorem Ipsum Salura
        </button>
      </div>
    </div>
  </div>
  <div
    className="closeButtonWrapper"
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      aria-label="Close"
      className="closeBtn button sizeMedium kindTertiary colorOnPrimaryColor noSidePadding"
      data-testid="alert-banner-close-button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      style={
        {
          "alignItems": "center",
          "height": "32px",
          "justifyContent": "center",
          "padding": 0,
          "width": "32px",
        }
      }
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon noFocusStyle"
        data-testid="icon"
        data-vibe="Icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 20 20"
        width="20"
      >
        <path
          d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`AlertBanner > should render with correctly with text and link 1`] = `
<div
  aria-label="banner"
  className="typography onPrimary start singleLineEllipsis text text2Normal alertBanner primary"
  data-testid="alert-banner"
  role="banner"
>
  <div
    className="content"
  >
    <div
      className="contentItem contentItemText"
    >
      <div
        className="ellipsis"
      >
        <div
          className="bannerText"
          data-testid="alert-banner-text"
        >
          <span>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
          </span>
        </div>
      </div>
    </div>
    <div
      className="contentItem"
    >
      <div
        className="marginLeft"
        data-testid="alert-banner-link"
      >
        <a
          aria-describedby=""
          aria-label=""
          aria-labelledby=""
          className="link colorPrimary"
          data-testid="link"
          href="https://monday.com"
          id=""
          onClick={[Function]}
          rel="noreferrer"
          target="_blank"
        >
          <span
            className="text bannerLink bannerLinkTextColorOnPrimary"
          >
            Lorem ipsum
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    className="closeButtonWrapper"
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      aria-label="Close"
      className="closeBtn button sizeMedium kindTertiary colorOnPrimaryColor noSidePadding"
      data-testid="alert-banner-close-button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      style={
        {
          "alignItems": "center",
          "height": "32px",
          "justifyContent": "center",
          "padding": 0,
          "width": "32px",
        }
      }
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon noFocusStyle"
        data-testid="icon"
        data-vibe="Icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 20 20"
        width="20"
      >
        <path
          d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
        />
      </svg>
    </button>
  </div>
</div>
`;
